### PR TITLE
React Native tutorial code

### PR DESCRIPTION
**Pull request by cbush**
This is the code for Phase 1 of the new Realm RN tutorial. It uses hooks and context to provide the authentication and data.

User registration does not seem to be available in the SDK yet, so it is not supported.

## JIRA
https://jira.mongodb.org/browse/DOCSP-9910

## Screenshots

Log in screen
image

Project (Tasks) screen 
image

Actions sheet after clicking an action
image

Prompt after clicking add task button
image

### CODE REVIEW

File: rn/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  root: true,
+  extends: '@react-native-community',
+};
```
Comment (terakilobyte) [inline rn/.eslintrc.js:4]:
```suggestion
  extends: '@react-native-community',
  rules: {
    'react-native/no-inline-styles': 'off',
  },
};
```

File: rn/TaskItem.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  if (task.status !== Task.STATUS_COMPLETE) {
+    actions.push({
+      title: 'Mark Complete',
+      action: () => {
+        setTaskStatus(task, Task.STATUS_COMPLETE);
+      },
+    });
+  }
+
+  // TODO: Add "rename" task action.
```
Comment (terakilobyte) [inline rn/TaskItem.js:55]: ??

File: rn/TasksProvider.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        realmRef.current = null;
+      }
+    };
+  }, [user, projectId]); // Declare dependencies list in the second parameter to useEffect().
+
+  // Define our create, update, and delete functions that users of the
+  // useTasks() hook can call.
+  const createTask = (newTaskName) => {
+    const realm = realmRef.current;
+    // TODO: Check that realm != null. If the realm is null, it isn't opened yet.
```
Comment (terakilobyte) [inline rn/TasksProvider.js:95]: ??

File: rn/TaskItem.js (hunk 2 above)
Comment (cbush) [reply rn/TaskItem.js:55]: Exercise for the reader

File: rn/TasksProvider.js (hunk 3 above)
Comment (cbush) [reply rn/TasksProvider.js:95]: Exercise for the reader

**Pull request merged**